src/e3/anod/driver.py

```diff
@@ -61,7 +61,28 @@ def activate(self, sandbox: SandBox, spec_repository: AnodSpecRepository) -> Non
                     kind=e.kind,
                     env=e.env(self.anod_instance, BaseEnv.from_env()),
                 )
-                self.anod_instance.deps[e.local_name] = dep_instance
+                if e.kind != "source":
```

Comment:
Hmmm, feel like this could be shared between the two files ? Maybe create a `@staticmethod` that you can call with parameters ?

`def get_source_dependency(e)` then call for `self.anod_instance.deps[e.local_name] = AnodContext.get_source_dependency(e)`

Reply:
What do you mean by `shared between the two files`?

Reply:
Well to me, the changes here and in `AnodContext` are exactly the same. So the code should be done in one place only ... Still, this is my point of view.

Reply:
No, because AnodDriver is used by the test suite, and is scheduled to be removed. Even if the code seems to be the same, AnodDriver doesn't do the same thing as AnodContext.

Reply:
Ok, but you have the error message to fix twice then ...

Reply:
Done.